src/main/java/org/gitlab4j/api/TopicsApi.java

```diff
@@ -0,0 +1,193 @@
+        return (response.readEntity(Topic.class));
+    }
+
+    /**
+     * Get all details of a topic as an Optional instance.
+     *
+     * <pre><code>GitLab Endpoint: GET /topics/:id</code></pre>
+     *
+     * @param id the topic ID
+     * @return the Topic for the specified topic id as an Optional instance
+     */
+    public Optional<Topic> getOptionalTopic(Integer id) {
+        try {
+            return (Optional.ofNullable(getTopic(id)));
+        } catch (GitLabApiException glae) {
+            return (GitLabApi.createOptionalFromException(glae));
+        }
+    }
+
+    /**
+     * Creates a new Topic. Available only for users who can create topics.
+     *
+     * <pre><code>GitLab Endpoint: POST /topics</code></pre>
+     *
+     * @param params a TopicParams instance holding the parameters for the topic creation
+     * @return the created Topic instance
+     * @throws GitLabApiException if any exception occurs
+     */
+    public Topic createTopic(TopicParams params) throws GitLabApiException {
+        Response response = post(Response.Status.CREATED, params.getForm(true), "topics");
+        return (response.readEntity(Topic.class));
+    }
+
+    /**
+     * Update a project topic.
+     *
+     * <pre><code>GitLab Endpoint: PUT /topics/:id</code></pre>
+     *
+     * @param id the topic id
+     * @param params a TopicParams instance holding the properties to Update
+     * @return the updated Topic instance
+     * @throws GitLabApiException at any exception
+     */
+    public Topic updateTopic(Integer id, TopicParams params) throws GitLabApiException {
+        Response response = putWithFormData(Response.Status.OK,
+            params.getForm(false), "topics",  id);
+        return (response.readEntity(Topic.class));
+    }
+
+    /**
+     * Uploads and sets the topic's avatar for the specified topic.
+     *
+     * <pre><code>GitLab Endpoint: PUT /topics/:id</code></pre>
+     *
+     * @param id the topic in the form of an Integer
+     * @param avatarFile the File instance of the avatar file to upload
+     * @return the updated Topic instance
+     * @throws GitLabApiException if any exception occurs
+     */
+    public Topic updateTopicAvatar(final Integer id, File avatarFile) throws GitLabApiException {
+        Response response = putUpload(Response.Status.OK, "avatar", avatarFile,  "topics", id);
+        return (response.readEntity(Topic.class));
+    }
+
+    /**
+     * Delete the topic's avatar for the specified topic.
+     *
+     * <pre><code>GitLab Endpoint: PUT /topics/:id</code></pre>
+     *
+     * @param id the topic in the form of an Integer
+     * @return the updated Topic instance
+     * @throws GitLabApiException if any exception occurs
+     */
+    public Topic deleteTopicAvatar(final Integer id) throws GitLabApiException {
+        Response response = putUpload(Response.Status.OK, "avatar", null,  "topics", id);
```

Comment:
This is failing here:

```
org.gitlab4j.api.GitLabApiException: Missing body part entity of type "application/octet-stream".
	at org.gitlab4j.api.AbstractApi.handle(AbstractApi.java:700)
	at org.gitlab4j.api.AbstractApi.putUpload(AbstractApi.java:560)
	at org.gitlab4j.api.TopicsApi.deleteTopicAvatar(TopicsApi.java:160)
	at ...
Caused by: javax.ws.rs.ProcessingException: Missing body part entity of type "application/octet-stream".
	at org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:312)
	at org.glassfish.jersey.client.JerseyInvocation.lambda$invoke$0(JerseyInvocation.java:662)
	at org.glassfish.jersey.client.JerseyInvocation.call(JerseyInvocation.java:697)
	at org.glassfish.jersey.client.JerseyInvocation.lambda$runInScope$3(JerseyInvocation.java:691)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:205)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:390)
	at org.glassfish.jersey.client.JerseyInvocation.runInScope(JerseyInvocation.java:691)
	at org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:661)
	at org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:439)
	at org.glassfish.jersey.client.JerseyInvocation$Builder.put(JerseyInvocation.java:328)
	at org.gitlab4j.api.GitLabApiClient.putUpload(GitLabApiClient.java:696)
	at org.gitlab4j.api.GitLabApiClient.putUpload(GitLabApiClient.java:677)
	at org.gitlab4j.api.AbstractApi.putUpload(AbstractApi.java:558)
	... 12 more
Caused by: java.lang.IllegalArgumentException: Missing body part entity of type "application/octet-stream".
	at org.glassfish.jersey.media.multipart.internal.MultiPartWriter.writeTo(MultiPartWriter.java:191)
	at org.glassfish.jersey.media.multipart.internal.MultiPartWriter.writeTo(MultiPartWriter.java:56)
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.invokeWriteTo(WriterInterceptorExecutor.java:242)
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.aroundWriteTo(WriterInterceptorExecutor.java:227)
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:139)
	at org.glassfish.jersey.message.internal.MessageBodyFactory.writeTo(MessageBodyFactory.java:1116)
	at org.glassfish.jersey.client.ClientRequest.doWriteEntity(ClientRequest.java:461)
	at org.glassfish.jersey.client.ClientRequest.writeEntity(ClientRequest.java:443)
	at org.glassfish.jersey.client.internal.HttpUrlConnector._apply(HttpUrlConnector.java:400)
	at org.glassfish.jersey.client.internal.HttpUrlConnector.apply(HttpUrlConnector.java:273)
	at org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:300)
	... 26 more
```

Reply:
In the TopicsApi Documentation it says "To remove a topic avatar, use a blank value for the avatar attribute." 

Im not 100% sure how all of your Code Conventions work, but maybe does this work?

`        Response response = post(Response.Status.OK,new GitLabApiForm().withParam("avatar",""),"topics",id);
`

Reply:
I have pushed 1cb014c7931bc6444c47cd4fef6a5d39de56a6d4